bindings/wasm/src/did/wasm_service.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity::did::ServiceEndpoint;
+use identity::iota::IotaDIDUrl;
+use identity::iota::IotaService;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsCast;
+
+use crate::did::WasmDIDUrl;
+use crate::error::Result;
+use crate::error::WasmResult;
+
+/// A DID Document Service used to enable trusted interactions associated
+/// with a DID subject.
+///
+/// See: https://www.w3.org/TR/did-core/#services
+#[wasm_bindgen(js_name = Service, inspectable)]
+pub struct WasmService(pub(crate) IotaService);
+
+#[wasm_bindgen(js_class = Service)]
+impl WasmService {
+  #[wasm_bindgen(constructor)]
+  pub fn new(service: IService) -> Result<WasmService> {
+    let id: IotaDIDUrl = service.id().into_serde().wasm_result()?;
+    let type_: String = service.type_();
+    let service_endpoint: ServiceEndpoint = deserialize_map_or_any(&service.service_endpoint())?;
+    let properties: Option<identity::core::Object> = deserialize_map_or_any(&service.properties())?;
+
+    IotaService::builder(properties.unwrap_or_default())
+      .id(id)
+      .type_(type_)
+      .service_endpoint(service_endpoint)
+      .build()
+      .map(WasmService::from)
+      .wasm_result()
+  }
+
+  /// Returns a copy of the `Service` id.
+  #[wasm_bindgen(getter)]
+  pub fn id(&self) -> WasmDIDUrl {
+    WasmDIDUrl::from(self.0.id().clone())
+  }
+
+  /// Returns a copy of the `Service` type.
+  #[wasm_bindgen(getter = type)]
+  pub fn type_(&self) -> String {
+    self.0.type_().to_owned()
+  }
+
+  /// Returns a copy of the `Service` endpoint.
+  #[wasm_bindgen(getter = serviceEndpoint)]
```

Comment:
I would use a method instead of a getter, but fine either way.

Reply:
I prefer getters here since there are no setters on this class and the fields are relatively simple (compared to DID Document, for instance).

We should decide on a convention and apply it everywhere though, since mixing getters and methods can be confusing.